src/ProductConstructionService/ProductConstructionService.WorkItems/WorkItemConfiguration.cs

```diff
@@ -47,14 +52,15 @@ public static async Task UseLocalWorkItemQueues(this IServiceProvider servicePro
         await queueClient.CreateIfNotExistsAsync();
     }
 
-    public static void AddWorkItemProcessors(this IServiceCollection services)
-    {
-        services.RegisterWorkItemProcessor<CodeFlowWorkItem, CodeFlowWorkItemProcessor>();
-    }
-
-    private static void RegisterWorkItemProcessor<TWorkItem, TProcessor>(this IServiceCollection services)
+    public static void AddWorkItemProcessor<TWorkItem, TProcessor>(this IServiceCollection services)
+        where TWorkItem : WorkItem
         where TProcessor : class, IWorkItemProcessor
```

Comment:
```suggestion
        where TProcessor : WorkItemProcessor<TWorkItem>
```

Reply:
I don't think we want this. It's more freedom for people to base their implementation on whatever they choose but they just need to implement the non-generic process themselves. So they can choose